v2.4 doc changes

manuwelakanade-GSLAB opened the pull request:
v2.4 doc changes

JosephPJordan commented on docs/rest_api2.rst:
```diff
@@ -0,0 +1,4259 @@
+|             |                                                                                                                                |                                                                                                                                                         |
+|             | 2. Right click the logical interface and click **Open in Object Store Browser**                                                |                                                                                                                                                         |
+|             |                                                                                                                                |                                                                                                                                                         |
+|             | 3. Click the > arrow of the dn property to see it’s children                                                                   |                                                                                                                                                         |
```
...its children

JosephPJordan replied:
...its children

JosephPJordan commented on docs/rest_api2.rst:
`````````````````diff
@@ -0,0 +1,4259 @@
+````````````````
+
++-------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Title             | Deletes a BIG-IP device from F5 ACI ServiceCenter Application (Note: The device is soft deleted, and once added back to the app, all data for the device is restored)   |
`````````````````
...from the F5 ACI ServiceCenter...

add 'the'

JosephPJordan commented on docs/rest_api2.rst:
```diff
@@ -0,0 +1,4259 @@
+|                    |                                                                       |
+|                    | Content: {error: Bad request}                                         |
++--------------------+-----------------------------------------------------------------------+
+| Notes              | Does not discovery BIG-IP VE and BIG-IP vCMP Guests                   |
```
Does not discover BIG-IP...

discover > discovery

JosephPJordan commented on docs/rest_api2.rst:
```diff
@@ -0,0 +1,4259 @@
+APIC managed objects can be accessed at https://<APIC_IP>/visore.html.
+
+Below is a table of such parameters and steps on how to retrieve them.
+These parameters will be required for L2-L3 Stitching tab.
```
These parameters will be required for the L2-L3 Stitching tab.

add "the"

JosephPJordan commented on docs/rest_api2.rst:
```diff
@@ -0,0 +1,4259 @@
+|                    |                                                                                                                                                                                                                                                          |
+|                    | Content: {error: Bad request}                                                                                                                                                                                                                            |
++--------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Notes              | As can be seen, the json provided has action "deploy”, and in the application, the json remains the same, except for the action attribute, which is changed to “dry-run” and a POST request is sent to <BIG-IP IP address>/mgmt/shared/appsvcs/declare   |
```
As you can see, the JSON provided has an action of **deploy**, and in the application, the JSON remains...

JosephPJordan commented on docs/rest_api2.rst:
``````````````````````````diff
@@ -0,0 +1,4259 @@
+`````````````````````````
+
++--------------------+-----------------------------------------------------------------------------------------------------------------------+
+| Title              | Submits AS3 declaration to specified BIG-IP device’s AS3 endpoint                                                     |
``````````````````````````
Submits the AS3 declaration to the AS3 endpoint on the specified BIG-IP device.

JosephPJordan commented on docs/rest_api2.rst:
```diff
@@ -0,0 +1,4259 @@
+|                    |                                                                                                                       |
+|                    | Content: {error: Bad request}                                                                                         |
++--------------------+-----------------------------------------------------------------------------------------------------------------------+
+| Notes              | The json from above dictionary is posted to the BIG IP AS3 endpoint <BIG-IP IP address>/mgmt/shared/appsvcs/declare   |
```
The JSON from the dictionary above...

JosephPJordan commented on docs/rest_api2.rst:
```````````````````````diff
@@ -0,0 +1,4259 @@
+``````````````````````
+
++--------------------+------------------------------------------------------------------------------------------------------------+
+| Title              | Get AS3 declaration JSON from given BIG-IP device                                                          |
```````````````````````
Get the AS3 declaration JSON from the given BIG-IP device

JosephPJordan commented on docs/rest_api2.rst:
```diff
@@ -0,0 +1,4259 @@
+|                    |                                                                                                            |
+|                    | Content: {error: Bad request}                                                                              |
++--------------------+------------------------------------------------------------------------------------------------------------+
+| Notes              | Get AS3 declaration from specified BIG-IP’s AS3 endpoint <BIG-IP IP address>/mgmt/shared/appsvcs/declare   |
```
Get the AS3 declaration from the specified...

JosephPJordan commented on docs/rest_api2.rst:
``````````````````````````diff
@@ -0,0 +1,4259 @@
+`````````````````````````
+
++--------------------+-------------------------------------------------------------------------+
+| Title              | Deletes an application from BIG-IP AS3 declaration                      |
``````````````````````````
....from the BIG-IP AS3 declaration

JosephPJordan commented on docs/rest_api2.rst:
```````````````````````````diff
@@ -0,0 +1,4259 @@
+``````````````````````````
+
++--------------------+-------------------------------------------------------------------------+
+| Title              | Get as3 async task response                                             |
```````````````````````````
Get AS3 async task response

JosephPJordan commented on docs/rest_api2.rst:
`````````````````diff
@@ -0,0 +1,4259 @@
+````````````````
+
++--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Title              | Gets Logical Device (L4-L7 device) information for specified Logical Device (Distinguished Name of Logical Device required)                                                 |
`````````````````
...for the specified Logical Device (Distinguished Name of the Logical Device is required)

JosephPJordan commented on docs/rest_api2.rst:
```diff
@@ -0,0 +1,4259 @@
+|                    |                                                                                                                                                                             |
+|                    | {                                                                                                                                                                           |
+|                    |                                                                                                                                                                             |
+|                    | "path": "Pod-1/Node-101/eth1/2",                                                                                                                        
```
...from a BIG-IP device

JosephPJordan replied:
...from a BIG-IP device

JosephPJordan replied:
..VLAN information for specific VLANs <or 'a specific VLAN'> from a BIG-IP Device to the F5 ACI...

JosephPJordan replied:
Creates a default gateway on a BIG-IP device

JosephPJordan replied:
Deletes a default gateway from a BIG-IP device

JosephPJordan replied:
...from a BIG-IP device to the F5 ServiceCenter...

JosephPJordan replied:
...from the F5 ACI ServiceCenter database to a BIG-IP device

JosephPJordan replied:
... on a BIG-IP device with a type of **vCMP Guest**

JosephPJordan replied:
VLANs

JosephPJordan replied:
a F5 BIG-IP <or just a BIG-IP>

JosephPJordan replied:
This is used for assigning and unassigning VLANs.  ... then it deletes the VLAN from the guest BIG-IP, otherwise

JosephPJordan replied:
If a Partition other than...

JosephPJordan replied:
If a Partition....

JosephPJordan replied:
from a BIG-IP device

JosephPJordan replied:
If a Partition other...

JosephPJordan commented on docs/faults.rst:
```diff
@@ -0,0 +1,32 @@
+
+1. Traverse to top-right corner menu.
+
+2. Click :guilabel:`App Faults`.
```
I'd combine steps one and two

1.  From the top-right corner menu, click :guilabel:`App Faults`.

JosephPJordan commented on docs/faults.rst:
```diff
@@ -0,0 +1,32 @@
+3. This displays a datagrid with the last 100 errors/warnings that occurred in the application’s background tasks and also displays resolution, if any.
+
+.. note::
+    - This specifically displays the faults in websocket and subscriptions for dynamic endpoint attach detach feature. 
```
...for the dynamic endpoint

JosephPJordan commented on docs/faults.rst:
```diff
@@ -0,0 +1,32 @@
+.. note::
+    - This specifically displays the faults in websocket and subscriptions for dynamic endpoint attach detach feature. 
+    
+    - To see older errors/warnings, please ssh to APIC server and check log files present in /data2/logs/F5Networks_F5ACIServiceCenter/faults/app.log
```
...errors/warnings, SSH to the APIC server and check the log files in /data2....

JosephPJordan commented on docs/faults.rst:
```diff
@@ -0,0 +1,32 @@
+
+1. Login to the BIG-IP.
+
+2. Click on any one of the feature tabs i.e. :guilabel:`Visibility` OR :guilabel:`L2-L3 Network Management` OR :guilabel:`L4-L7 App Services`.
```
Click any one of the...

- on

JosephPJordan commented on docs/faults.rst:
```diff
@@ -0,0 +1,32 @@
+
+2. Click on any one of the feature tabs i.e. :guilabel:`Visibility` OR :guilabel:`L2-L3 Network Management` OR :guilabel:`L4-L7 App Services`.
+
+3. Click on the :guilabel:`View Faults` icon.
```
Click the ...

JosephPJordan commented on docs/faults.rst:
```diff
@@ -0,0 +1,32 @@
+.. note::
+    - The datagrid also displays the type of error (Visibility, L2-L3 Network Management OR L4-L7 App Services. It is possible to filter the errors by type)
+    
+    - To see older errors/warnings, please ssh to APIC server and check log files present in /data2/logs/F5Networks_F5ACIServiceCenter/faults/<BIG-IP IP>.log
```
... errors/warnings, SSH to the APIC server

JosephPJordan commented on docs/faq.rst:
```diff
@@ -42,6 +42,12 @@ All F5 ACI ServiceCenter operations in-turn perform REST API calls to BIG-IP or
+
+**Q. I’m able to see only the last 100 errors/warnings in 'View Faults'. How can I see older faults for my BIG-IP? (View Faults feature is available in versions 2.4+)**
+
+The F5 ACI ServiceCenter UI will show the last 100 errors or warnings which were observed on a particular BIG-IP. To check the older errors/warnings, please ssh to the APIC server which has the app container. The faults logs for BIG-IPs are available at location data2/logs/F5Networks_F5ACIServiceCenter/faults.
```
..errors/warnings, SSH to the APIC... 

....are available at data2/logs (remove 'location').

JosephPJordan commented on docs/navigate.rst:
```diff
@@ -31,6 +31,7 @@ View VIP table
    -  If a node is not operational on the APIC, it is not displayed in the VIP table.
    -  If a pool is empty and does not have any nodes, a pool entry is not displayed in the VIP table.
    -  If a VIP does not have an assigned default pool, the VIP is not displayed in the VIP table.
+   -  If any of the node members have an associated FQDN, it will be displayed in FQDN column (Supported in v2.4+).
```
... in the FQDN column

JosephPJordan commented on docs/navigate.rst:
```diff
@@ -47,6 +48,7 @@ View Node table
    -  If a specific node is not operational on the APIC, it isn't displayed in the Node table.
    -  If a pool does not have any nodes, the pool is not displayed in any of the entries in the Node table.
    -  If a VIP does not have an assigned default pool, the VIP is not displayed in any of the entries in the Node table.
+   -  If any of the nodes have an associated FQDN, it will be displayed in FQDN column (Supported in v2.4+).
```
...in the FQDN column

JosephPJordan commented on docs/release-notes.rst:
```diff
@@ -1,3 +1,141 @@
+
+**F5 ACI ServiceCenter Upgrade is not supported from Version 1.0 to Version 2.4**
+
+The F5 ACI ServiceCenter application does not have upgrade support from Version 1.0 to Version 2.4. In order to install a new version of the app the steps to be followed are:
```
In order to install a new version of the app, use the following steps:

JosephPJordan commented on docs/release-notes.rst:
```diff
@@ -1,3 +1,141 @@
+2. User may not find previously added BIG-IP entries in the application
+3. New Application installation may fail
+
+**Workaround**: Wait for 20 minutes and try to access the application again. The Glusterfs should recover within some time automatically and the application should be accessible again.
```
I'd suggest getting rid of "within some time".. so it would be 'should recover automatically'

JosephPJordan commented on docs/release-notes.rst:
```diff
@@ -1,3 +1,141 @@
+
+**F5 ACI ServiceCenter HA clusters show IPs even if the devices are added using hostnames**
+
+For BIG-IP in HA cluster, when one BIG-IP with hostname is added to F5 ACI ServiceCenter, the IP corresponding to that hostname also gets added in the side menu. 
```
For BIG-IP devices in a HA cluster
or 
For a BIG-IP device in a HA cluster...

JosephPJordan commented on docs/release-notes.rst:
```diff
@@ -1,3 +1,141 @@
+
+For BIG-IP in HA cluster, when one BIG-IP with hostname is added to F5 ACI ServiceCenter, the IP corresponding to that hostname also gets added in the side menu. 
+
+**Workaround:** Delete the device which got added using IP by clicking on the X icon besides the IP of the host. Repeat the same process for the peer of this device as well - now both devices should be seen inside the cluster with hostname.  
```
...by clicking the delete (x) icon next to the IP of the host.

suggest removing "as well"

JosephPJordan commented on docs/release-notes.rst:
```diff
@@ -1,3 +1,141 @@
+
+**Floating IP auto sync and Default Gateway auto sync will not work when hosts are added in an HA cluster using hostnames**
+
+**Workaround:** Manually sync the Floating IPs and Default Gateway to Application DB by clicking on “Sync To DB”
```
...Gateway to the Application DB by clicking **Sync To DB**.

JosephPJordan commented on docs/release-notes.rst:
```diff
@@ -1,3 +1,141 @@
+
+getldevinfo.json and createbigipvlan.json APIs will show an error of the type “Invalid DN <someDn>, wrong rn prefix <somePrefix> at position X/Y“
+
+Root-cause: During vlan creation using createbigipvlan.json API, the VLAN table in the F5 ACI ServiceCenter saves VLAN database entries. One of the fields in the VLAN table is the lifDn which is the Distinguished name of Logical Interface (in the Logical device) on APIC. If during App REST API automation, anyone creates a VLAN using createbigipvlan.json and enters invalid string in lifDn paramete ...
```
During VLAN creation...

JosephPJordan commented on docs/release-notes.rst:
```diff
@@ -1,3 +1,141 @@
+
+**Error on EPG mapping delete operation**
+
+When a dynamic endpoint mapping is added to an application using the "Manage Endpoint Mappings", the application gets created on BIG-IP. If this mapping is deleted using the "RESET" button on "Manage Endpoint Mappings" form, user may encounter an error "The requested Pool Member (/Partition/App/Pool /NodePartition/NodeIP) was not found"
```
...to an application using "Manage Endpoint Mappings", the application gets created on the BIG-IP.

JosephPJordan commented on docs/release-notes.rst:
```diff
@@ -1,3 +1,141 @@
+
+When a dynamic endpoint mapping is added to an application using the "Manage Endpoint Mappings", the application gets created on BIG-IP. If this mapping is deleted using the "RESET" button on "Manage Endpoint Mappings" form, user may encounter an error "The requested Pool Member (/Partition/App/Pool /NodePartition/NodeIP) was not found"
+
+**Workaround:** Click on the "Submit" button again, and the mapping will be reset properly without any errors. 
```
Click the **Submit** button again...

JosephPJordan commented on docs/release-notes.rst:
```diff
@@ -1,3 +1,141 @@
+
+**AS3 application can either have static nodes or dynamic nodes but not both**
+
+AS3 applications will support either static nodes or dynamic nodes (Using Manage Endpoint Mappings button) but not both. 
```
...nodes (using the **Manage Endpoint Mappings** button)

JosephPJordan commented on docs/release-notes.rst:
```diff
@@ -1,3 +1,141 @@
+If an APIC endpoint group has more than 60 endpoints attached, then the endpoint list will not get reflected on the BIG-IP, and users may experience service restarts on BIG-IP. 
+
+**Workaround:** For any dynamic endpoint mapping, please ensure that the number of endpoints in the corresponding APIC endpoint group never crosses 60.
+
```
... group never exceeds 60.

JosephPJordan commented on docs/release-notes.rst:
```diff
@@ -1,3 +1,141 @@
+
+**Node not removed from BIG-IP pool when node IP is a substring of some other node's IP**
+
+If a node, for example node with IP 1.2.3.4 is deleted from APIC, and there is also another node 1.2.3.40 of which the original IP is a substring, it may be possible that dynamic end point attach detach feature is not able to delete 1.2.3.4 from BIG-IP. Note: The pool members will get deleted as expected. 
```
If a node, for example a node with an IP of 1.2.3.4, is deleted...

..it may be possible that the dynamic endpoint attach feature...

JosephPJordan commented:
@manuwelakanade-GSLAB and @payalsin , made some comments in the files (let me know if there's another or better way to do this).  Let me know when it's complete and I'll approve the request.

JosephPJordan commented:
Also, @manuwelakanade-GSLAB, there was one file i think was edited twice, but the comments showed up in the same place.  If something doesn't line up, just ignore it or shoot me a question.

manuwelakanade-GSLAB commented:
Thanks @JosephPJordan : I will address the  comments soon and update the pull request

manuwelakanade-GSLAB commented:
Hi @JosephPJordan : I have addressed all the comments: https://github.com/F5Networks/f5-aci-servicecenter/pull/71/files. Please let me know if there are any other changes needed. Thanks.

JosephPJordan commented on docs/faq.rst:
```diff
@@ -170,9 +170,9 @@ User may observe the error: "Failed to get a new subscription. Subscription Refr
+**Q. How can I change the management port of a BIG-IP device which is already added in the F5 ACI ServiceCenter?**
 
-Click the delete (X) icon besides the BIG-IP to delete it. Re-add the BIG-IP to F5 ACI ServiceCenter with the changed port. The BIG-IP data will still be retained after the delete and re-add.
+Click the delete (X) icon besides the BIG-IP to delete it. Re-add the BIG-IP to F5 ACI ServiceCenter with the changed port (For example, from the default 443 to 8443). The BIG-IP data will still be retained after the delete and re-add.
```
either "beside" or "next to", not besides

manuwelakanade-GSLAB replied:
Fixed

JosephPJordan commented on docs/release-notes.rst:
```diff
@@ -25,21 +25,21 @@ Symptoms users may see:
+For BIG-IP devices in an HA cluster, when one BIG-IP with hostname is added to F5 ACI ServiceCenter, the IP corresponding to that hostname also gets added in the side menu. 
 
-**Workaround:** Delete the device which got added using IP by clicking on the X icon besides the IP of the host. Repeat the same process for the peer of this device as well - now both devices should be seen inside the cluster with hostname.  
+**Workaround:** Delete the device which got added using IP by clicking on the delete (X) icon besides the IP of the host. Repeat the same process for the peer of this device - now both devices should be seen inside the cluster with hostname.  
```
same here, "beside" or "next to", not besides

JosephPJordan commented:
Nice work @manuwelakanade-GSLAB , just noticed two small things that were missed.  If you can change those, I'll approve the pull.  Thanks.

manuwelakanade-GSLAB commented:
Hi @JosephPJordan : I have addressed the 2 new comments. Thanks.

manuwelakanade-GSLAB commented:
Thanks for merging the request @JosephPJordan
